### PR TITLE
Use raw objects, not pointers, for Firefly classes.

### PR DESCRIPTION
**Pull request by ademuri**
This is less error prone, since it relies on the natural object lifecycle.

### CODE REVIEW

File: lib/effect/Effect.cpp [hunk 1]
```diff
@@ -12,50 +12,53 @@ uint8_t Effect::GetThresholdSin(int16_t x, uint8_t threshold) {
...
-    // Solid color
-    {{HUE_RED, 255, 255}},
-    {{HUE_ORANGE, 255, 255}},
-    {{HUE_YELLOW, 255, 255}},
-    {{HUE_GREEN, 255, 255}},
-    {{HUE_AQUA, 255, 255}},
-    {{HUE_BLUE, 255, 255}},
-    {{HUE_PURPLE, 255, 255}},
-    {{HUE_PINK, 255, 255}},
+const std::vector<ColorPalette> Effect::palettes() {
```
Comment (wmmiii) [inline lib/effect/Effect.cpp:18]: I don't understand the reasoning behind this change.

File: src/devices/fancy-node/fancy-node.cc [hunk 2]
```diff
@@ -27,10 +27,10 @@ static const DeviceDescription *SimpleRfBoardDescription(
 const DeviceDescription *const test_device =
     SimpleRfBoardDescription(1, {Bright});
 
-RadioHeadRadio *radio;
-NetworkManager *nm;
-FastLedManager *led_manager;
-RadioStateMachine *state_machine;
+RadioHeadRadio *radio = new RadioHeadRadio();
```
Comment (wmmiii) [inline src/devices/fancy-node/fancy-node.cc:30]: Why not just store the object here instead of a pointer to the heap?

Comment (wmmiii) [reply src/devices/fancy-node/fancy-node.cc:30]: (Is this related to the comment on 
https://github.com/candykingdom/firefly2/pull/51/files#diff-8c7980a9c0f5f1aafdf3541eebda36f02aed6bade3088cec21f82dec7f53bd0dR17

File: lib/effect/Effect.cpp (hunk 1 above)
Comment (ademuri) [reply lib/effect/Effect.cpp:18]: Added a comment explaining this.

File: src/devices/fancy-node/fancy-node.cc (hunk 2 above)
Comment (ademuri) [reply src/devices/fancy-node/fancy-node.cc:30]: Updated that other comment - node crashes if `RadioHeadRadio` is statically allocated, but I'm not sure why.

**Review by wmmiii — APPROVED** [review]

**Pull request merged**